server/src/main/java/org/elasticsearch/env/NodeEnvironment.java

```diff
@@ -494,10 +495,25 @@ private void maybeLogHeapDetails() {
     private static NodeMetaData loadOrCreateNodeMetaData(Settings settings, Logger logger,
                                                          NodePath... nodePaths) throws IOException {
         final Path[] paths = Arrays.stream(nodePaths).map(np -> np.path).toArray(Path[]::new);
+
+        final Set<String> nodeIds = new HashSet<>();
+        for (final Path path : paths) {
+            final NodeMetaData metaData = NodeMetaData.FORMAT.loadLatestState(logger, NamedXContentRegistry.EMPTY, path);
```

Comment:
Previously, a multi-path setup would succeed loading node state, as long as one of the paths were readable. With this change, I think we will fail if any of the paths are not readable/corrupt? See MetaDataStateFormat.loadGeneration(...)

Reply:
This is true, and I prefer the added strictness, but I've marked this as `team-discuss` so we can all make sure we're in agreement.